### DISCUSSION TITLE
Stimuli alternation and background color

### DISCUSSION THREAD

**Discussion by ShaBeNou** [expyriment/expyriment#221] [Q&A]
Hello :)

I tried building a simple code where some instrcutions are given ("Please hold still as experiment will soon begin"), the system waits for a trigger ("5"),plays a video and then another line of instructions appears.

For some reason, the video presentation does not replace the text and my screen begins flashing with the text showing behind. 

Using txt1.unload(), txt1.clear_surface(0 and loading a blank screen on top of it didn't work.

Also, it seems like even though I defined the background color of the screen to be gray using `exp=xpy.design.Experiment(name="Pilot", background_colour=(211,211,211))` the video is loading on black background (hence what causes the flashing)

Is it possible to show a video on the background color of the experiemnt? How can I remove text once I'm done with it?

The code as whole is attached below:

```
import expyriment as xpy

xpy.misc.download_from_stash(content="extras")

# Initialize experiment
exp = xpy.design.Experiment(name="Pilot", background_colour=(211,211,211))
xpy.control.initialize(exp)

# Set defaults
xpy.control.defaults.open_gl=2
SCAN_TRIGGER = xpy.misc.constants.K_5
SCAN_TRIGGER_LTP_ADDRESS = None  # None = USB (keyboard emulation)

# Load stimuli
# Instructions
txt1 = xpy.stimuli.TextLine(text="Please hold still as experiment will soon begin", text_size=40, text_colour=(255,255,255))
txt1.preload()

# Video
#v = xpy.stimuli.Video("b1-a.mp4")
v = xpy.stimuli.Video("hedgehog.mp4")
xpy.control.stop_audiosystem()
v.preload()

# Further Instructions
txt2 = xpy.stimuli.TextLine(text="Please hold still and wait for instructions", text_size=40, text_colour=(255,255,255))
txt2.preload()

# Start experiment
xpy.control.start(skip_ready_screen=True)

txt1.present()
trigger = exp.keyboard  # For USB (keyboad emulation) trigger
#trigger = xpy.io.TriggerInput(io.ParallelPort(SCAN_TRIGGER_LTP_ADDRESS))  # For LTP trigger
trigger.wait(SCAN_TRIGGER)

txt1.unload()

v.play()
v.present()
video_presentation_time = exp.clock.time
v.wait_end()
v.stop()

txt2.present()
exp.clock.wait(5000)

xpy.control.end(goodbye_delay=1000, goodbye_text="Saving...")
```

P.S.
I tried setting up the default color of all screens (such as the logo and subject id screens) to be grey with no success either. 

Thanks ahead!
Shahar

**Comment by fladd** [comment]
Is this still an issue with Expyriment 1.0, where the Video stimulus has been completely overhauled?

**Comment by chrplr** [comment]
> Is this still an issue with Expyriment 1.0

Yes, I replicate the phenomenon here (linux/expyriment 1.0.0)


--
Christophe Pallier <http://www.pallier.org>
Directeur de  Recherche CNRS
EMR CNRS 9003 & INSERM-CEA  U992
Cognitive Neuroimaging Lab <http://www.unicog.org/> | CNRS
<https://www.cnrs.fr/en/le-cnrs/missions> | INSERM
<https://www.inserm.fr/en/about-us/inserm-at-a-glance/> | CEA
<https://www.cea.fr/english/Pages/cea/the-cea-a-key-player-in-technological-research.aspx>
| Université Paris-Saclay
<https://www.universite-paris-saclay.fr/en/about/about-universite-paris-saclay>

**Reply by fladd** [reply]
Ah, I see now what the issue is. I made the Video stimulus such that it doesn't have a background, in order for smaller videos to be overlayed on other things. In your script, your custom background is only applied once when you present the txt1. But since you are using a double buffer, the back buffer is still without background colour.

For now you can solve this by presenting a BlankScreen twice before showing the video.

For the future, we should think about how the Video stimulus should handle this:
(1) by getting a "background" argument
or 
(2) by implementing the clear/update arguments of the present function of the other stimuli to the present function of the Video stimulus (although then you would need to still take care of the double buffer yourself, by calling this twice...).
